### PR TITLE
Add support to save safetensors checkpoint directly into onnx

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
Currently, when `torch.onnx.dynamo_export` is called within `torch.onnx.enable_fake_mode`, all the external pytorch checkpoint files used to initialize the model are automatically and used by `torch.onnx.ONNXProgram.save` to recreate the initializers for
the newly exported ONNX model.

This API extends the mechanism for HuggingFace models that use safetensors weights. This PR detects safetensors state files and converts them to PyTorch format using mmap on a temporary file, which is deleted after conversion is finished.

Without this PR, the user would have to convert the safetensors files to pytorch format manually and feed it to `torch.onnx.ONNXProgram.save` manually.

### CODE REVIEW

File: torch/onnx/_internal/fx/serialization.py
Comment (BowenBao) [inline torch/onnx/_internal/fx/serialization.py:153]: why (load from safetensor) -> (save as tmp torch.pt) -> (load from tmp torch.pt) -> state_dict,
but not (load from safetensor) -> state_dict?

Comment (thiagocrepaldi) [reply torch/onnx/_internal/fx/serialization.py:153]: ouch, the reason is not in this pr by accident lol

by having a temorary file, we can mmap it and support large models even when the safetensor doesnt fit in memory. will update the pr

Comment (BowenBao) [reply torch/onnx/_internal/fx/serialization.py:153]: But this step `(load from safetensor) -> (save as tmp torch.pt)` already requires fully loading weights into memory + more overhead of saving it to tmp file, or am I missing something?

Comment (thiagocrepaldi) [reply torch/onnx/_internal/fx/serialization.py]: I will double check how the Storage differs when the tensor comes from disk or is created in memory.
I will change to safetensors -> state_dict to get unblocked in the mean time

**Review by BowenBao — APPROVED** [review]

**Comment by thiagocrepaldi** [comment]
Unrelated XLA failure
```bash
++ cat ../.github/ci_commit_pins/xla.txt
+ git checkout 8f913829abd9de749339d8d74b7357e1be3a7907
fatal: reference is not a tree: 8f913829abd9de749339d8d74b7357e1be3a7907
```

**Comment by BowenBao** [comment]
We'd need extra approval for files outside of onnx cc @malfet 

The unrelated test is marked flaky in trunk so it won't block merging.

**Comment by thiagocrepaldi** [comment]
@malfet @kit1980 could you help with this one liner log change to a file outside onnx folder?

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**
